lib/Debugger.ProtocolHandler/DebuggerImpl.cpp

```diff
@@ -470,6 +470,61 @@ namespace JsDebug
         }
     }
 
+    SkipPauseRequest DebuggerImpl::EvaluateConditionOnBreakpoint(int bpId)
+    {
+        if (bpId < 0)
+        {
+            return SkipPauseRequest::RequestNoSkip;
+        }
+
+        DebuggerBreakpoint *bp = nullptr;
+        auto it = m_breakpointMap.begin();
+        while (it != m_breakpointMap.end())
+        {
+            if (it->second.GetActualId() == bpId)
+            {
+                bp = &it->second;
+                break;
+            }
+            it++;
+        }
+
+        if (bp == nullptr)
+        {
+            return SkipPauseRequest::RequestNoSkip;
+        }
+
+        try
+        {
+            String condition = bp->GetCondition();
+            if (condition.length() != 0)
+            {
+                DebuggerCallFrame callFrame = m_debugger->GetCallFrame(0);
+                std::unique_ptr<ExceptionDetails> exceptionDetails;
+                auto exprReturned = callFrame.Evaluate(condition, true, &exceptionDetails);
+                // Ignoring the exception information.
+
+                // If the condition is provided, the debugger will stop only when the expression is evaluated to true
+                String defaultValue("");
+                if (exprReturned->getType() == String("boolean") 
+                    && exprReturned->hasDescription() 
+                    && exprReturned->getDescription(defaultValue) == String("true"))
```

Comment:
Does it have to be boolean type "true" or can it be any "truthy" value for the expression?

Reply:
It will be for any truthy value. When we evaluate the expression, the ChakraCore will determine the type of return of the expression and convert to Boolean and set that to returning object.

Reply:
I would probably not use that Evaluate call, it's designed specifically for returning to the debugger frontend.  You really just want to call JsDiagEvaluate on call frame zero, coerce the result to boolean, then get the bool from it.